config/decode.go

```diff
@@ -20,12 +20,20 @@ func processUnusedConfigKeys(md mapstructure.Metadata, file string) error {
 	err := fmt.Errorf("'%s' has invalid keys: %s", file, strings.Join(md.Unused, ", "))
 
 	for _, key := range md.Unused {
-		if key == "provider" {
-			err = fmt.Errorf(`%s
-	'provider' is an invalid key for Consul Terraform Sync configuration, try 'terraform_provider'.
-	terraform_provider configuration blocks are similar to provider blocks in Terraform but have additional features supported only by Consul Terraform Sync.
-
-`, err)
+		switch key {
+		case "provider":
+			return fmt.Errorf(`%s
+
+'provider' is an invalid key for Consul Terraform Sync configuration, try
```

Comment:
This doesn't have to do with your changes, but I've noticed we have some discrepancies in our output where we sometimes have `Consul Terraform Sync` instead of `Consul-Terraform-Sync`.

Reply:
thanks for the callout, i'll go ahead and update this occurrence to be consistent with hyphens.